src/main/java/seedu/address/model/company/Company.java

```diff
@@ -23,6 +28,27 @@
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Company {
+    //Comparator for filterInternships
+    private static final Comparator<Internship> internshipComparator = (internship1, internship2) -> {
+        // Extract the interview date if it exists
+        Optional<LocalDateTime> date1 = internship1.getInternshipDateTime()
+                .map(InternshipInterviewDateTime::getInternshipDateTime);
+        Optional<LocalDateTime> date2 = internship2.getInternshipDateTime()
+                .map(InternshipInterviewDateTime::getInternshipDateTime);
+
+        // Check if both have no dates
+        if (date1.isEmpty() && date2.isEmpty()) {
+            return 0; // Both have no dates, leave them unsorted.
+        } else if (date1.isEmpty()) {
+            return 1; // Only internship2 has a date, place internship1 at the back.
+        } else if (date2.isEmpty()) {
+            return -1; // Only internship1 has a date, place internship2 at the back.
+        } else {
+            // Both internships have dates, compare based on the interview date.
+            return date1.get().compareTo(date2.get());
```

Comment:
Does this change the behaviour of the existing internship list? If so,  I don't think we can change it due to the feature freeze :(